argo/providers/kfp/provider.go

```diff
@@ -91,6 +92,13 @@ func (kfpp KfpProvider) DeletePipeline(ctx context.Context, providerConfig KfpPr
 		Context: ctx,
 	}, nil)
 
+	if err != nil {
+		var errorResult *pipeline_service.DeletePipelineDefault
```

Comment:
I like how this has been done for run schedules: https://github.com/sky-uk/kfp-operator/pull/336/files#diff-b10da3f39a3b0dc026835d1b8510861f3b8188618d9e35b550b80bd8cc4191cbR421

Reply:
Theres no common interface between the two error types we try to handle so a generic function is not possible even with constraints.  In order to move the very small amount of common code into a common function will not be of much use past having an expressive name.